src/WebJobs.Script.WebHost/Middleware/RestoreRawRequestPathMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    /// <summary>
+    /// Middleware to restore the raw request URL path received, in the current request URL.
+    /// App service front end decodes encoded strings in the request URL path.
+    /// This causes routing to fail if the route param value has %2F in it.
+    /// Refer below issues for more context:
+    /// https://github.com/dotnet/aspnetcore/issues/40532#issuecomment-1083562919
```

Comment:
Is it important to mention these github issues here? I think the comment you added above explains the purpose and we can also add these links to the PR description. Though it's fine if there is specific reason for them to be here.

Reply:
I don't think it hurts to have additional context available via the GH issue link, might get lost otherwise